src/sentry/lang/native/appconnect.py

```diff
@@ -0,0 +1,237 @@
+"""Integration of native symbolication with Apple App Store Connect.
+
+Sentry can download dSYMs directly from App Store Connect, this is the support code for
+this.
+"""
+
+import enum
+import io
+import logging
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+import jsonschema
+import requests
+
+from sentry.lang.native.symbolicator import APP_STORE_CONNECT_SCHEMA
+from sentry.models import Project
+from sentry.utils import json
+from sentry.utils.appleconnect import appstore_connect, itunes_connect
+
+logger = logging.getLogger(__name__)
+
+
+# The key in the project options under which all symbol sources are stored.
+SYMBOL_SOURCES_PROP_NAME = "sentry:symbol_sources"
+
+
+# The symbol source type for an App Store Connect symbol source.
+SYMBOL_SOURCE_TYPE_NAME = "appStoreConnect"
+
+
+class InvalidCredentialsError(Exception):
+    """Invalid credentials for the App Store Connect API."""
+
+    pass
+
+
+class InvalidConfigError(Exception):
+    """Invalid configuration for the appStoreConnect symbol source."""
+
+    pass
+
+
+class NoDsymsError(Exception):
+    """No dSYMs were found."""
+
+    pass
+
+
+@enum.unique
+class BuildKind(enum.Enum):
+    ALL = 1
+    PRE_RELEASE = 2
+    RELEASE = 3
+
+
+@dataclass(frozen=True)
+class BuildInfo:
+    """Information about an App Store Connect build.
+
+    A build is identified by the tuple of (app_id, platform, version, build_number), though
+    Apple mostly names these differently.
+    """
+
+    # The kind of build, either PRE_RELEASE or RELEASE
+    kind: BuildKind
+
+    # The app ID
+    app_id: str
+
+    # A platform identifying e.g. iOS, TvOS etc.
+    #
+    # These are not always human readable but some opaque string supplied by apple.
+    platform: str
+
+    # The human-readable version, e.g. "7.2.0".
+    #
+    # Each version can have multiple builds, Apple naming is a little confusing and calls
+    # this "bundle_short_version".
+    version: str
+
+    # The build number, typically just a monotonically increasing number.
+    #
+    # Apple naming calls this the "bundle_version".
+    build_number: str
+
+
+def get_app_store_config(project: Project, config_id: str) -> json.JSONData:
```

Comment:
it might be better to return a more specific type here as we probably assume the data is in a valid format (we do the same in the frontend: assume specific types for API responses without validating them, as it's a "trusted" source). Is this `Config` again?